scripts/dc2/create_lensed_host_ic.py

```diff
@@ -0,0 +1,176 @@
+import os
+import numpy as np
+import pandas as pd
+import argparse
+from sqlalchemy import create_engine
+from lsst.sims.catUtils.utils import ObservationMetaDataGenerator
+from desc.sims.GCRCatSimInterface import get_obs_md
+from lsst.sims.utils import angularSeparation
+from dc2_utils import instCatUtils
+
+__all__ = ['hostImage']
+
+class hostImage(instCatUtils):
+    """Takes FITS stamps and includes them in instance catalogs
+    hostImage takes the following arguments:
+    ra_center: Right ascension of the center of the field (in degrees)
+    dec_center: Declination of the center of the field (in degrees)
+    fov: Field-of-view angular radius (in degrees).  2 degrees will cover
+        the LSST focal plane."""
+
+    def __init__(self, obs_md, fov):
+
+        self.ra = obs_md.pointingRA
+        self.dec = obs_md.pointingDec
+        self.radius = fov
+        self.bandpass = obs_md.bandpass
+        self.obs_md = obs_md
+
+        self.bandpass_lookup = {'u': 0, 'g': 1, 'r': 2, 'i': 3, 'z': 4, 'y': 5}
+
+
+    def format_catalog(self, df_line, fits_file_name, image_dir):
+        """
+        Formats the output instance catalog to include entries for the FITS 
+        stamps produced by generate_lensed_host_***.py
+		
+		Parameters: 
+		-----------
+		df_line: string
+			pandas data frame line
+		fits_file_name: string
+			the filename of the FITS stamp
+		image_dir: string 
+			the location of the FITS stamps 
+		Returns:
+		-----------
+		cat_str: a string containing the line of parameters to go into an instance file
+        """
+
+        fits_info = fits_file_name.split('_')
+        lens_id = np.int(fits_info[0])
+        sys_magNorm_list = np.array(fits_info[1:-1], dtype=np.float)
+        sys_magNorm = sys_magNorm_list[self.bandpass_lookup[self.bandpass]]
+        gal_type = fits_info[-1].split('.')[0]
+
+        # galaxy_id = np.right_shift(lens_id, 10)
+        # galaxy_id *= 10000
+        # galaxy_id += 4*df_line['twinkles_system']
+        # sys_id = np.left_shift(galaxy_id, 10)
+        # if gal_type == 'bulge':
+        #     sys_id += 97
+        # elif gal_type == 'disk':
+        #     sys_id += 107
+        # else:
+        #     raise ValueError('Invalid Galaxy Component Type in filename')
+        sys_id = lens_id
+
+        cat_str = 'object %i %f %f %f %s %f 0 0 0 0 0 %s 0.01 0 CCM %f %f CCM %f %f\n' % (sys_id,
```

Comment:
Using `sys_id = lens_id` here seems to yield disk and bulge components that have the same object ID for a given lensed system, but that field needs to be unique for each of the rendered objects.  Since imSim saves the object IDs as part of the checkpointing data so that it knows which ones to skip when restarting from a checkpoint file, imSim will only render the first one it encounters.  If we make the ID field a string, we could just append `'b'` for bulge and `'d'` for disk to make these IDs unique.

Reply:
Good idea. Added it into the code and cleaned up the commented lines.